src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -466,7 +472,7 @@ public void doKill() {
         execution = null; // ensures isInProgress returns false
         FlowInterruptedException suddenDeath = new FlowInterruptedException(Result.ABORTED);
         finish(Result.ABORTED, suddenDeath);
-        executionPromise.setException(suddenDeath);
+        getSettableExecutionPromise().setException(suddenDeath);
```

Comment:
No `isDone` check here?

Reply:
Good question -- per the JavaDocs: "This method will return {@code true} if the exception was
   * successfully set, or {@code false} if the future has already been set or
   * cancelled."